chemprop/utils.py

```diff
@@ -145,6 +146,122 @@ def load_checkpoint(path: str,
 
 
+def overwrite_state_dict(loaded_param_name: str,
+                        model_param_name: str,
+                        loaded_state_dict: collections.OrderedDict,
+                        model_state_dict: collections.OrderedDict,
+                        logger: logging.Logger = None) -> collections.OrderedDict:
+    """
+    Overwrites a given parameter in the current model with the loaded model.
+    :param loaded_param_name: name of parameter in checkpoint model.
+    :param model_param_name: name of parameter in current model.
+    :param loaded_state_dict: state_dict for checkpoint model.
+    :param model_state_dict: state_dict for current model.
+    :param logger: A logger.
+    :return: The updated state_dict for the current model. 
+    """
+    debug = logger.debug if logger is not None else print
+
+    
+    if model_param_name not in model_state_dict:
+        debug(f'Pretrained parameter "{model_param_name}" cannot be found in model parameters.')
+        
+    elif model_state_dict[model_param_name].shape != loaded_state_dict[loaded_param_name].shape:
+        debug(f'Pretrained parameter "{loaded_param_name}" '
+              f'of shape {loaded_state_dict[loaded_param_name].shape} does not match corresponding '
+              f'model parameter of shape {model_state_dict[model_param_name].shape}.')
+    
+    else:
+        debug(f'Loading pretrained parameter "{model_param_name}".')
+        model_state_dict[model_param_name] = loaded_state_dict[loaded_param_name]    
+    
+    return model_state_dict
+
+def load_frzn_model(model: torch.nn,
+                    path: str,
+                    current_args: Namespace = None,
+                    cuda: bool = None,
+                    logger: logging.Logger = None) -> MoleculeModel:
+    """
+    Loads a model checkpoint.
+    :param path: Path where checkpoint is saved.
+    :param current_args: The current arguments. Replaces the arguments loaded from the checkpoint if provided.
+    :param cuda: Whether to move model to cuda.
+    :param logger: A logger.
+    :return: The loaded MoleculeModel.
+    """
+    debug = logger.debug if logger is not None else print
+
+    loaded_mpnn_model = torch.load(path, map_location=lambda storage, loc: storage)
+    loaded_state_dict = loaded_mpnn_model['state_dict']
+    loaded_args = loaded_mpnn_model['args']
+
+    model_state_dict = model.state_dict()
+    
+    if loaded_args.number_of_molecules==1 & current_args.number_of_molecules==1:      
+        encoder_param_names = ['encoder.encoder.0.W_i.weight', 'encoder.encoder.0.W_h.weight', 'encoder.encoder.0.W_o.weight', 'encoder.encoder.0.W_o.bias']
+        if current_args.checkpoint_frzn is not None:
+            # Freeze the MPNN
+            for param_name in encoder_param_names:
+                model_state_dict = overwrite_state_dict(param_name,param_name,loaded_state_dict,model_state_dict)
+            
+        if current_args.frzn_ffn_layers > 0:         
```

Comment:
I ran some tests for what happens when you specify a larger number of frozen ffn than you have layers. If you set number of layers >= your number of layers, you get an error when it tries to do gradient descent but nothing has saved gradients. Maybe raise an error somewhere in here that has a descriptive message about the number of ffn layers